### PR TITLE
CR for Job #20255 - Visually show the grouping of scripts in the "Running Scripts" dialog

### PR DESCRIPTION
**Pull request by thoys**
Code Review for Job #20255 - Workitem available at https://www.worklist.net/20255

### CODE REVIEW

**Comment by murillodigital** [comment]
test this please

**Comment by murillodigital** [comment]
test this please

**Comment by thoys** [comment]
anything wrong here @murillodigital  ?

**Comment by murillodigital** [comment]
nope @thoys your PR is my guinea pig for some other work I'm doing, sorry for the spam.

**Comment by thoys** [comment]
hehe thats ok

File: interface/src/ScriptsModel.cpp [hunk 1]
```diff
@@ -170,7 +208,7 @@ bool ScriptsModel::parseXML(QByteArray xmlFile) {
                     xml.readNext();
                     lastKey = xml.text().toString();
                     if (jsRegex.exactMatch(xml.text().toString())) {
-                        _remoteFiles.append(new ScriptItem(lastKey.mid(MODELS_LOCATION.length()), S3_URL + "/" + lastKey));
+                        _treeNodes.append(new TreeNodeScript(lastKey.mid(MODELS_LOCATION.length()), S3_URL + "/" + lastKey, SCRIPT_ORIGIN_REMOTE));
                     }
                 }
                 xml.readNext();
```
Comment (AndrewMeadows) [inline interface/src/ScriptsModel.cpp]: I don't see where _treeNodes' contents are deleted when the ScriptsModel is deleted.  Is this a real memory leak or is there some special circumstance that I'm not seeing here?

I went back to the original code in upstream/master and it looks like there were pre-existing similar memory leaks in _remoteFiles and _localFiles.

Comment (thoys) [inline interface/src/ScriptsModel.cpp]: ah yes I see, I'm on it, sorry.

**Comment by thoys** [comment]
Test this please

**Comment by thoys** [comment]
test this please

**Pull request merged**